### DISCUSSION TITLE
Long runing process, start sharp

### DISCUSSION THREAD

**Discussion by Nikola7i** [serenity-is/Serenity#5328] [Help]
Hi, I need help about situation when I need to call new dialog with date  picker before batch process. Date should be passed to batch record.
1. Pick Date Range
2. Call batch process 
3. Use Date range variables in batch

Anybody have some example ?

Regards,

**Comment by VictorTomaili** [comment]
Please use discussion for question related topics.

**Comment by edwardch** [comment]
@Nikola7i I have done this before, and a send this long running process to Hangfire to do it for me. See [Hangfire Background Jobs wiki](https://github.com/serenity-is/Serenity/wiki/Background-Tasks:-Using-Hangfire-Background-Jobs) for info on how to integrate Hangfire.

From there you can create an endpoint that then enqueues the job with something like `Hangfire.BackgroundJob.Enqueue<YourJob>(job => job.DoSomething(request.StartDate, request.EndDate))`.

Then create a form with parameters (like your start and end dates). 
In your form dialog you can call something like 
```ts
YourService.QueueTheJob({
    StartDate: this.form.StartDate.value,
    EndDate: this.form.EndDate.value
}, response => {
    Q.notifySuccess("Successfully triggered");

    this.dialogClose();
});
```

Hope that helps!

**Reply by Nikola7i** [reply]
First, tnx for help. :-)

I use ServiceResponceAction to start long running process in endpoint. The code call stored procedure and need to pass start and end date .
That all work fine because I use today date (new date()).

I need to call form (is way to put local date on grid?)  before call  action in  grid.ts when press button , and pass data to endpoint.

This is button to call 

   buttons.push({
                title: 'Obračunaj amortizaciju ',
                cssClass: 'send-button',
                onClick: () => {


                    if (!this.onViewSubmit()) {
                        return;
                    }

                    var keys = this.rowSelection.getSelectedAsInt32();
                    if (!keys.length) {
                        Q.notifyError("Molim označiti OS!");
                        return;
                    }



                    Q.confirm("Obračunaj amortizaciju " + keys.length + " na označenim OS?", () => {
                        var operationID = "myOperationID_" + new Date().getTime()


                        OsnovnaSredstva.OsnovnoSredstvoService.**Action**({
                            OrderIDs: keys,
                            OperationID: operationID,
                            DatumOD: new Date(),
                            DatumDO: new Date()


                        },
                            response => {
                                Q.notifySuccess("Obračun završen.");
                            }, {
                            blockUI: false,
                            onCleanup: () => {
                                progressDialog && progressDialog.dialogClose();
                                progressDialog = null;
                                operationID = null;
                            }
                        });

                        var progressDialog = new BasicProgressDialog();
                        progressDialog.dialogOpen();
                        progressDialog.max = 100;
                        progressDialog.value = 0;

                        function updateProgress() {
                            if (!operationID)
                                return;

                            if (progressDialog && progressDialog.cancelled) {
                                OsnovnaSredstva.OsnovnoSredstvoService.Cancel({
                                    OperationID: operationID
                                }, response => {
                                    progressDialog.dialogClose();
                                    progressDialog = null;
                                    return;
                                });

                                operationID = null;
                                return;
                            }

                            OsnovnaSredstva.OsnovnoSredstvoService.Status({
                                OperationID: operationID
                            }, r => {
                                if (!progressDialog)
                                    return;
                                progressDialog.value = Math.floor(r.PercentComplete);
                                progressDialog.title = r.StatusText;
                                window.setTimeout(() => updateProgress(), 500);
                            });
                        }


                        window.setTimeout(() => updateProgress(), 500);
                    });


                }

            })
